### ISSUE TITLE
googlesignin exception

### ISSUE THREAD

**Issue by namdroid** [firebase/mlkit-custom-image-classifier#6]
i have copied google-service.json to app folder:

Now i get exception at line in emulator and device:
GoogleSignInAccount googleUser = await _googleSignIn.signInSilently();

Exception has occurred.
PlatformException (PlatformException(sign_in_required, com.google.android.gms.common.api.ApiException: 4: 4: , null))

**Comment by viju85** [comment]
I get a similar error. 

PlatformException(sign_in_failed, com.google.android.gms.common.api.ApiException: 12500: , null)

Have been trying several things mentioned on stackoverflow for the same. But still no luck. Any help is appreciated.

**Comment by skotwani** [comment]
Did you set the SHA-1 for your app? See more  here: https://github.com/flutter/flutter/issues/25640#issuecomment-449589417

**Comment by viju85** [comment]
Yes I tried that and ensured SHA-1 is added to the firebase console. Still it doesn't work. Anyone has this working?

**Comment by namdroid** [comment]
i have setted own package name, generate new SHA-1, but no luck

**Comment by prakhar1989** [comment]
@viju85 @namdroid Make sure you create a new app on Firebase with the new package name you've decided to keep. While creating the app, add the SHA1 by running the following ([more docs](https://developers.google.com/android/guides/client-auth))

```
 keytool -list -v -alias androiddebugkey -keystore ~/.android/debug.keystore
```
The password is `android`. Then copy/paste the SHA1 into the settings page in Firebase for your app. Lastly, remember to rebuild and reinstall the app.

**Comment by namdroid** [comment]
@prakhar1989 thanks, but of course i did this step before like in other apps.

**Comment by VickySalunkhe** [comment]
hi @namdroid did find any solution ? 
I am also facing the same issue.

**Comment by viju85** [comment]
@VickySalunkhe @namdroid I finally got it working by following this solution.

"You need to add support gmail in settings of firebase"
https://stackoverflow.com/a/56286513/341617. 

Hope this helps.

**Comment by prakhar1989** [comment]
Thanks @viju85 for pointing it out! I'll add that part to the getting started steps.
